### ISSUE TITLE
Add "normalize" option to Search input?

### ISSUE THREAD

**Issue by Fil** [observablehq/inputs#135]
This would allow to search texts with accents (more generally diacritics) by normalizing everything (input and values) with a common function (besides string.toLowerCase()).

Here's a poc https://observablehq.com/@fil/search-normalize; it takes a lot of code since I had to basically copy all of https://github.com/observablehq/inputs/blob/main/src/search.js, just to add a call to string.normalize here and there.

It's in use in https://observablehq.com/@visionscarto/aires-d-accueil-les-donnees

**Comment by easz** [comment]
BTW, it might be not enough with `normalize` if we want to support all kind of language.

for example the current spec would just not be suitable for CJK languages, since there is not really space between characters.
> The default filter splits the current query into space-separated tokens and checks that each token matches the beginning of at least one string in the data’s columns, case-insensitive.

At the moment crafting a customize `filter` is probably the only way to implement for various language.

**Comment by mbostock** [comment]
Right, my expectation was that you would provide your own _filter_ option to control this behavior in the general. But I’m happy to hear ideas on how we can either expose hooks in the existing searchFilter implementation, and a _normalize_ hook as a preprocessing step that defaults to `x => x.toLowerCase()` sounds reasonable.

**Comment by easz** [comment]
one quick idea: `Search` can expose (or accept) a termFilter and an optional function to transform data getting compared.

something like
~~~
Search(data, {
    query: query,
    matchingFilter: customTermFilter, // a user defined termFilter
    transformData: customTransformData // function to transform source data
  });
~~~
https://observablehq.com/@easz/search-in-various-languages#here

**Comment by jrus** [comment]
@Fil there is some relevant discussion at https://stackoverflow.com/q/990904

Apparently one trick that will strip most diacritics is `str.normalize('NFD').replace(/[\u0300-\u036f]/g, '')`

**Comment by Fil** [comment]
this may be solved by https://github.com/observablehq/inputs/pull/216

**Comment by mootari** [comment]
@Fil #216 has been merged. Can this issue be closed?

**Comment by Fil** [comment]
No. #216 supports non-ascii chars (so that searching for "île" in ["Ile", "île"] will find the second word). This issue is about supporting normalization (searching for "île" would find the two words—which is what I expect as a user).
